testing/steps/test_step_cloner_for_each_data_input.py

```diff
@@ -18,53 +18,47 @@
 
 def test_step_cloner_should_transform():
     tape = TapeCallbackFunction()
-    p = Pipeline([
-        StepClonerForEachDataInput(
+    p = StepClonerForEachDataInput(
             Pipeline([
                 FitCallbackStep(tape),
                 MultiplyByN(2)
             ])
         )
-    ])
     data_inputs = _create_data((2, 2))
 
     processed_outputs = p.transform(data_inputs)
 
-    assert isinstance(p[0].steps[0], Pipeline)
-    assert isinstance(p[0].steps[1], Pipeline)
+    assert isinstance(p.steps[0], Pipeline)
+    assert isinstance(p.steps[1], Pipeline)
     assert np.array_equal(processed_outputs, data_inputs * 2)
 
 
 def test_step_cloner_should_fit_transform():
     tape = TapeCallbackFunction()
-    p = Pipeline([
-        StepClonerForEachDataInput(
+    p = StepClonerForEachDataInput(
             Pipeline([
                 FitCallbackStep(tape),
                 MultiplyByN(2)
             ])
         )
-    ])
     data_inputs = _create_data((2, 2))
     expected_outputs = _create_data((2, 2))
 
     p, processed_outputs = p.fit_transform(data_inputs, expected_outputs)
 
-    assert isinstance(p[0].steps[0], Pipeline)
-    assert isinstance(p[0].steps[1], Pipeline)
+    assert isinstance(p.steps[0], Pipeline)
+    assert isinstance(p.steps[1], Pipeline)
```

Comment:
are you sure this test is complete? The tape seems unused in the asserts. I would have thought that the Tape did something.